openwisp_radius/api/views.py

```diff
@@ -359,6 +359,8 @@ def post(self, request, *args, **kwargs):
                     organization=Organization.objects.get(pk=self.request.auth),
                 )
                 batch = RadiusBatch(**options)
+                batch.full_clean()
+                batch.save()
```

Comment:
It wasn't working because batch wasn't saved before it was used to create a ManyToMany relationship with users in `batch.prefix_add(prefix, number_of_users)` function.

But this raises more questions than it answers:
Why were the automated not able to find this error, that's what I'll have to do now, and fix tests accordingly!

Reply:
Okay, I thought that the testcase was busted but that's not the case.
The error line: `self.users.add(user)` simply doesn't throw an error when used during testing and when request is made from admin.

Now I am in a situation where the patch works but I don't know why.
It's a dangerous situation, so please test it to see if you can replicate the `IntegrityError` when these lines are removed and API request to `BatchView` from prefix user generation is made.